src/lib/support/jsontlv/JsonToTlv.cpp

```diff
@@ -0,0 +1,418 @@
+    }
+    else if (strcmp(elementType, kElementTypeUInt) == 0)
+    {
+        type.tlvType = TLV::kTLVType_UnsignedInteger;
+    }
+    else if (strcmp(elementType, kElementTypeBool) == 0)
+    {
+        type.tlvType = TLV::kTLVType_Boolean;
+    }
+    else if (strcmp(elementType, kElementTypeFloat) == 0)
+    {
+        type.tlvType  = TLV::kTLVType_FloatingPointNumber;
+        type.isDouble = false;
+    }
+    else if (strcmp(elementType, kElementTypeDouble) == 0)
+    {
+        type.tlvType  = TLV::kTLVType_FloatingPointNumber;
+        type.isDouble = true;
+    }
+    else if (strcmp(elementType, kElementTypeBytes) == 0)
+    {
+        type.tlvType = TLV::kTLVType_ByteString;
+    }
+    else if (strcmp(elementType, kElementTypeString) == 0)
+    {
+        type.tlvType = TLV::kTLVType_UTF8String;
+    }
+    else if (strcmp(elementType, kElementTypeNull) == 0)
+    {
+        type.tlvType = TLV::kTLVType_Null;
+    }
+    else if (strcmp(elementType, kElementTypeStruct) == 0)
+    {
+        type.tlvType = TLV::kTLVType_Structure;
+    }
+    else if (strncmp(elementType, kElementTypeArray, strlen(kElementTypeArray)) == 0)
+    {
+        type.tlvType = TLV::kTLVType_Array;
+    }
+    else
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+bool isUnsignedInteger(const std::string & s)
+{
+    size_t len = s.length();
+    if (len == 0)
+    {
+        return false;
+    }
+    for (size_t i = 0; i < len; i++)
+    {
+        if (!isdigit(s[i]))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool isSignedInteger(const std::string & s)
+{
+    if (s.length() == 0)
+    {
+        return false;
+    }
+    if (s[0] == '-')
+    {
+        return isUnsignedInteger(s.substr(1));
+    }
+    return isUnsignedInteger(s);
+}
+
+bool isValidBase64String(const std::string & s)
+{
+    const std::string base64Chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+    size_t len                    = s.length();
+
+    // Check if the length is a multiple of 4
+    if (len % 4 != 0)
+    {
+        return false;
+    }
+
+    size_t paddingLen = 0;
+    if (s[len - 1] == '=')
+    {
+        paddingLen++;
+        if (s[len - 2] == '=')
+        {
+            paddingLen++;
+        }
+    }
+
+    // Check for invalid characters
+    for (char c : s.substr(0, len - paddingLen))
+    {
+        if (base64Chars.find(c) == std::string::npos)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+struct ElementContext
+{
+    std::string jsonName;
+    TLV::Tag tag = TLV::AnonymousTag();
+    ElementTypeContext type;
+    ElementTypeContext subType;
+};
+
+bool CompareByTag(const ElementContext & a, const ElementContext & b)
+{
+    return TLV::TagNumFromTag(a.tag) < TLV::TagNumFromTag(b.tag);
+}
+
+CHIP_ERROR ParseJsonName(const std::string name, ElementContext & elementCtx)
```

Comment:
Is this really parsing a "name"?  Or is it parsing a JSON object key?

Reply:
@bzbarsky-apple We had this discussion related to Json naming convention. The official rfc8259 calls it name/value pair. Some Json libraries call it key/value but we decided to follow an official definition and I updated this code to call it "name" everywhere.